be1-go/hub/organizer.go

```diff
@@ -484,3 +486,203 @@ func (c *laoChannel) processRollCallObject(data message.Data) error {
 
 	return nil
 }
+func (c *laoChannel) processElectionObject(msg message.Message) error {
+	action := message.ElectionAction(msg.Data.GetAction())
+
+	if action != message.ElectionSetupAction {
+		return &message.Error{
+			Code:        -1,
+			Description: fmt.Sprintf("invalid action: %s", action),
+		}
+	}
+	err := c.createElection(msg)
+	if err != nil {
+		return xerrors.Errorf("failed to setup the election %v", err)
+	}
+
+	return nil
+}
+
+func (c *laoChannel) createElection(msg message.Message) error {
+	o := c.hub
+
+	o.Lock()
+	defer o.Unlock()
+
+	// Check the data
+	data, ok := msg.Data.(*message.ElectionSetupData)
+	if !ok {
+		return &message.Error{
+			Code:        -4,
+			Description: "failed to cast data to SetupElectionData",
+		}
+	}
+
+	// Check if the Lao ID of the message corresponds to the channel ID
+	encodedLaoID := base64.StdEncoding.EncodeToString(data.LaoID)
+	channelID := c.channelID[6:]
+	if channelID != encodedLaoID {
+		return &message.Error{
+			Code:        -4,
+			Description: fmt.Sprintf("Lao ID of the message (Lao: %s) is different from the channelID (channel: %s)", encodedLaoID, channelID),
+		}
+	}
+
+	// Compute the new election channel id
+	encodedElectionID := base64.StdEncoding.EncodeToString(data.ID)
```

Comment:
I recall there was an issue with using StdEncoding and "/". Did we decide to use `base64.URLEncoding`? @flmichel

Reply:
Oh yeah that's what was specified in the issue [#281](https://github.com/dedis/student_21_pop/issues/281) I just saw the issue and fixed this